src/Repositories/RedisMasterSupervisorRepository.php

```diff
@@ -81,9 +81,10 @@ public function get(array $names)
 
             return ! $record[0] ? null : (object) [
                 'name' => $record[0],
-                'pid' => $record[1],
-                'status' => $record[2],
-                'supervisors' => json_decode($record[3], true),
+                'environment' => $record[1],
+                'pid' => $record[2],
+                'status' => $record[3],
+                'supervisors' => json_decode($record[4], true),
```

Comment:
Does this not break existing Horizon installs who have the records saved in a different sequence?

Reply:
I checked this scenario specifically, where I started a mastersupervisor and then switched to this branch. HMGET returns the keys as requested in the order at line 75. [For any keys that don't exist, null is returned](https://redis.io/commands/hmget/). As there might be master supervisors already running with missing environments, the key might return null which will be properly handled as the fallback environment will be used in the masterSuperVisorController.

Reply:
I mean, doesn't the order of the keys and values get switched? I think the mapping of these values will be wrong?

Reply:
I can put the environment variable at the end as well, but according to the [redis docs](https://redis.io/commands/hmget/) it doesn't matter:
> **Return** [Array reply](https://redis.io/docs/reference/protocol-spec#resp-arrays): list of values associated with the given fields, in the same order as they are requested.

> For every field that does not exist in the hash, a nil value is returned.

Reply:
I think I see what you mean. Seems good to me!